### PR TITLE
Filter arrays when sanitizing passed data

### PR DESCRIPTION
**Pull request by fvovan**
follow up of #2238

### CODE REVIEW

**Comment by fvovan** [comment]
@fauvel please review.
@zazabe fyi

**Comment by fauvel** [comment]
Could be more elegant (e.g. recursive implementation of `_sanitizeUtf`), but lgtm!

File: library/CM/Http/Request/Abstract.php [hunk 1]
```diff
@@ -320,7 +320,18 @@ public function setUri($uri) {
         $querySanitized = [];
         foreach ($query as $key => $value) {
             $key = self::_sanitizeUtf($key);
-            $querySanitized[$key] = self::_sanitizeUtf($value);
+
+            if (is_array($value)) {
+                array_walk_recursive($value, function (&$innerValue) {
+                    if (is_string($innerValue)) {
```
Comment (fauvel) [inline library/CM/Http/Request/Abstract.php]: If you're still not sure that it must be a string at this stage... maybe you'd better use a safe, recursive implementation for `_sanitizeUtf`!

Comment (fvovan) [inline library/CM/Http/Request/Abstract.php]: i'm almost 100% sure it is a string, but added it just to double check, 
i'll test it without checking condition a bit later.

Comment (njam) [inline library/CM/Http/Request/Abstract.php]: What happens if we pass numbers? Will they be string or integer here?

Comment (fvovan) [inline library/CM/Http/Request/Abstract.php]: They will be casted to strings by `mb_parse_url()`

**Pull request merged**